### PR TITLE
expansion/hygiene: Some renaming, refactoring and comments

### PR DESCRIPTION
**Pull request by petrochenkov**
Pure refactoring, no functional changes.
Commits are isolated and self-descriptive.

### CODE REVIEW

**Comment by abonander** [comment]
Thanks for including me but I never understood the hygiene system very well anyway.

**Comment by petrochenkov** [comment]
@oli-obk 
Root is its own parent so we can avoid `Option`s.
Transparency for the root also doesn't make much sense, so we can choose it arbitrarily. I'll try to add some asserts making sure that it's not inspected from outside of `hygiene.rs`.
The root check in `Mark::modern` is subsumed by the opaqueness check because the root is now opaque.

**Comment by petrochenkov** [comment]
Updated.

**Comment by petrochenkov** [comment]
+ one more refactoring commit

**Comment by oli-obk** [comment]
This PR is going to murder clippy xD

@bors r+

File: src/libproc_macro/quote.rs [hunk 1]
```diff
@@ -85,9 +85,6 @@ impl ProcMacro for Quoter {
                    _: ::syntax_pos::Span,
                    stream: tokenstream::TokenStream)
                    -> tokenstream::TokenStream {
-        let mut info = cx.current_expansion.mark.expn_info().unwrap();
-        info.callee.allow_internal_unstable = true;
-        cx.current_expansion.mark.set_expn_info(info);
```
[the remark this answers is not in the document]
Comment (eddyb) [reply src/libproc_macro/quote.rs]: Nevermind, I see that `SyntaxExtension::ProcMacro` got an extra field.

**Pull request merged**